units/ssh_keys_update_local.go

```diff
@@ -0,0 +1,77 @@
+package units
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/evergreen-ci/evergreen"
+	"github.com/mongodb/amboy"
+	"github.com/mongodb/amboy/dependency"
+	"github.com/mongodb/amboy/job"
+	"github.com/mongodb/amboy/registry"
+	"github.com/pkg/errors"
+)
+
+const localUpdateSSHKeysJobName = "update-ssh-keys-local"
+
+type localUpdateSSHKeysJob struct {
```

Comment:
This job is quite risky. We should come up with a deployment strategy. We could lock ourselves out of every app server and build host.

Reply:
The greatest risk is probably if you happen to overwrite our current SSH key that's the key maintained by Chef. I added more sanity checking to `(*evergreen.Settings) ValidateAndDefault()` so we can be sure that updating the SSH keys won't delete or modify existing keys (including the Chef SSH key).

This job runs every time an app server restarts so if for some reason the local SSH key gets deleted, we can just restart and it'll be back after this job runs.